operator/pkg/helmreconciler/apply.go

```diff
@@ -178,6 +181,18 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured) error {
 	}
 
 	gvk := obj.GetObjectKind().GroupVersionKind()
+
+	// check minor version only
```

Comment:
What is the impact for users on older versions? It feels weird to have a conditional like this. Will 1.15 users be broken, but only in edge cases?

Also, do we need any of the features of SSA beta2 (k8s 1.18)?

Reply:
for users on 1.15 or prior we still use the existing customized apply logic, and it has issues in some cases, for example the operator does not preserve the fields set by server, we should fix that separately, for example a recent fix related: https://github.com/istio/istio/pull/28072

Reply:
for beta 2 update: https://github.com/istio/istio/pull/28072, seems like it is update on server side, it does not affect how we configure the controller

Reply:
> and conflicts for these objects cannot be resolved. With Kubernetes 1.18, all new objects will have the managedFields attached to them and provide accurate information on conflicts.

So we don't care since we always force conflicts anyways, right?

Reply:
that is true